sympy/polys/rings.py

```diff
@@ -2531,20 +2531,336 @@ def compose(f, x, a=None):
 
         return poly
 
-    # TODO: following methods should point to polynomial
-    # representation independent algorithm implementations.
+    def coeff_wrt(self, x, deg):
+        """
+        Coefficient of ``self`` with respect to ``x**deg``.
+
+        Treating *self* as a univariate polynomial in ``x`` this finds the
+        coefficient of ``x**deg`` as a polynomial in the other generators.
+
+        Parameters
+        ==========
+
+        x : generator or generator index
+            The generator or generator index to compute the expression for.
+        deg : int
+            The degree of the monomial to compute the expression for.
+
+        Returns
+        =======
+
+        :py:class:`~.PolyElement`
+            The coefficient of ``x**deg`` as a polynomial in the same ring.
+
+        Examples
+        ========
+
+        >>> from sympy.polys import ring, ZZ
+        >>> R, x, y, z = ring("x, y, z", ZZ)
+
+        >>> p = 2*x**4 + 3*y**4 + 10*z**2 + 10*x*z**2
+        >>> deg = 2
+        >>> p.coeff_wrt(2, deg) # Using the generator index
+        10*x + 10
+        >>> p.coeff_wrt(z, deg) # Using the generator
+        10*x + 10
+        >>> p.coeff(z**2) # shows the difference between coeff and coeff_wrt
+        10
+
+        See Also
+        ========
+
+        coeff, coeffs
+
+        """
+        p = self
+        i = p.ring.index(x)
+        terms = [(m, c) for m, c in p.iterterms() if m[i] == deg]
+
+        if not terms:
+            return p.ring.zero
+
+        monoms, coeffs = zip(*terms)
+        monoms = [m[:i] + (0,) + m[i + 1:] for m in monoms]
+        return p.ring.from_dict(dict(zip(monoms, coeffs)))
+
+    def prem(self, g, x=None):
+        """
+        Computes the pseudo-remainder of the polynomial ``self`` with respect
+        to ``g``.
+
+        Euclidean division is a well-defined operation for univariate
+        polynomials ``f`` and ``g`` with coefficients in a field. It yields a
+        unique pair of polynomials ``q`` and ``r``, the quotient and remainder,
+        respectively, satisfying ``f = gq + r``, where ``deg(r) < deg(g)``.
```

Comment:
I don't think that we need to duplicate this definition here given that it is already in `pdiv`. We can just say
```
        Pseudo-remainder of the polynomial ``self`` with respect to ``g``.

        The pseudo-quotient ``q`` and pseudo-remainder ``r`` with respect to
        ``z`` when dividing ``f`` by ``g`` satisfy ``mf = gq + r``,
        where ``deg(r,z) < deg(g,z)`` and
        ``m = LC(g,z)**(deg(f,z) - deg(g,z)+1)``.

        See :meth:`pdiv` for explanation of pseudo-division.
```
Probably
```
``mf = gq + r``
```
should be changed to
```
``m*f = g*q + r``
```
or
```
``m f = g q + r``
```
or otherwise it should be changed to LaTeX.  When not in LaTeX format it is not clear that `mf` is a product of `m` and `f` rather than a single variable name.

In the examples section it would be good to show an example that contrasts `pquo` with `quo` to show the difference. There should be a see also for `quo`.